sylvia-derive/src/message.rs

```diff
@@ -609,11 +803,13 @@ impl<'a> GlueMessage<'a> {
                     let val = #sylvia ::serde_value::Value::deserialize(deserializer)?;
 
                     #(#deserialization_attempts)*
+                    #impl_deserialization_attempt
 
                     Err(D::Error::custom(format!(
-                        "Expected any of {} messages, but cannot deserialize to neither of those\n{}",
+                        "Expected any of {} or {} messages, but cannot deserialize to neither of those\n{}",
                         stringify!(#(#messages_type),*),
-                        [#(#deser_errors,)*].join("\n")
+                        stringify!(#impl_msg_name),
+                        [#(#deser_errors,)* #impl_deser_errors].join("\n")
```

Comment:
Why didnt you just add a message to a variant array as I suggested? Seems like IT world reduce generated code a bit.

Reply:
https://github.com/CosmWasm/sylvia/pull/24#discussion_r957698212

I'm pretty sure that it's the same as in your suggestion. Please take a look once more.

Reply:
Yea, it is the same. I was misreading on phone. Sorry.